feat: add retrieval state to retrieval events

kylehuntsman opened the pull request:
Adds retrieval event state to the subscribable retrieval events.

It was a little difficult to add state to the events because there isn't really a shared context between all the different retrieval functions. It would be a bit better if the idea of a `Retrieval` was a separate context that was created, encapsulating the idea of a single retrieval. Then it can hold state context. I felt that was too big of a lift for right now, but talk of separating storage and retrieval functionality in the future could include this work.

hannahhoward commented on filclient.go:
```diff
@@ -449,7 +449,7 @@ func (fc *FilClient) GetAsk(ctx context.Context, maddr address.Address) (*networ
 		// publish fail event, log the err
 		retrievalEvent := rep.RetrievalEvent{Code: rep.RetrievalEventFailure, Status: fmt.Sprint("failed connecting to miner:", err)}
-		fc.retrievalEventPublisher.Publish(retrievalEvent)
+		fc.retrievalEventPublisher.Publish(retrievalEvent, rep.RetrievalState{})
```
whoa. Just realized a major issue here. this is the wrong place for retrieval events! This is the storage ask protocol, not the retrieval protocol. These are in the wrong place. I should have caught before but just realized now. The method these should be in is the `RetrievalQuery` & `RetrievalQueryToPeer` method, not this one!

hannahhoward reviewed (changes requested):
Hey so, we've got our query events in the wrong place, which is part of why we have so little state to work with. This was already there from the last PR, but it needs to be fixed prior to merge.

kylehuntsman commented:
> Hey so, we've got our query events in the wrong place, which is part of why we have so little state to work with. This was already there from the last PR, but it needs to be fixed prior to merge.

Moved event dispatches to the correct functions. Also added `StorageProviderAddr` to the `RetrievalState` so that we can record either the SP's peer ID or miner address since the `RetrievalQuery` function only provides an address while the others use peer IDs.

kylehuntsman commented:
I pushed a commit that logs the retrieval events using a different logger instance. This way logging filclient retrieval events can be made explicit in the use case you don't actually want the retrieval event log spam, or you _only_ want the retrieval event log spam.

hannahhoward commented:
LGTM

elijaharita commented on rep/event.go:
```diff
@@ -24,10 +25,13 @@ type RetrievalEvent struct {
-	Finished      time.Time
+	PayloadCid          *cid.Cid
+	PieceCid            *cid.Cid
+	StorageProviderId   *peer.ID
```
nit: `ID` instead of `Id` used everywhere else in filclient

elijaharita commented on rep/event.go:
```diff
@@ -24,10 +25,13 @@ type RetrievalEvent struct {
-	PeerId        peer.ID
-	ClientAddress peer.ID
-	Finished      time.Time
+	PayloadCid          *cid.Cid
```
out of curiosity, why all pointers?
